docs/api/decorators/prop.md

```diff
@@ -73,6 +73,22 @@ class Defaulted {
 }
 ```
 
+You may also set the `default` schema option to a function. Mongoose will execute that function and use the return value as the default. 
+
+Example: 
+
+```ts
+class Defaulted {
+  @prop({ 
+    default (this: DocumentType<Defaulted>) {
```

Comment:
```suggestion
    default: (this: DocumentType<Defaulted>) => {
```

from my knowledge, the one before was not valid JS/TS

Reply:
it is.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions

Reply:
its just a shorthand for the same thing

Reply:
learned a thing, thanks for providing source on this one, thought i would prefer keeping it in the same style as the other code for now